### ISSUE TITLE
Usage of Motion Sensors with InAppLocalHostServer

### ISSUE THREAD

**Issue by aatmmr** [pichillilorenzo/flutter_inappwebview#838]
First of all, thank you for creating this package which is highly useful and flexible.

We would like to run a simple HTML5 game in the WebView which is loaded via `InAppLocalHostServer`. The game uses motion sensors as user controls which works prior to iOS 14 without any issues. Since iOS 14, the user must consent the sensor usage. Using `permission_handler` requesting the constent works but does not apply to the game running on the localhost. If we add a separate permission request to the game itself, an alert will show up asking:

"localhost" Would Like To Access Motion and Orientation.

Giving consent to the usage will make the game work - so far so good.

However, for a better UX, is it possible to:

- Change "localhost" or the consent request to a custom text?
- Pass the consent somehow to the localhost instance, such as, for camera or microphone as described in [package docs](https://inappwebview.dev/docs/enable-camera-for-html-inputs/)?

Or am I missing something?

The app runs on package version `5.3.0` and flutter `stable` branch.

**Comment by kuhnroyal** [comment]
I also can't get motion events to work for a page server from a local file. But I think this is related to `InAppWebView` and not `InAppLocalHostServer`.

There is this https://bugs.webkit.org/show_bug.cgi?id=203287 saying that the `WKUIDelegate` needs to be set, but I think it is set.

There are also these new delegate methods for iOS 15: https://developer.apple.com/documentation/webkit/wkuidelegate?changes=latest_minor&language=objc

**Comment by kuhnroyal** [comment]
Hacking this into `InAppWebView.swift`:

```swift
    @available(iOS 15.0, *)
    public func webView(_ webView: WKWebView,
           requestDeviceOrientationAndMotionPermissionFor origin: WKSecurityOrigin,
                initiatedByFrame frame: WKFrameInfo,
                 decisionHandler: @escaping (WKPermissionDecision) -> Void) {
         decisionHandler(.grant)
     }
 ```

And combining it with some javascript after user interaction seems to work:

```javascript
    DeviceMotionEvent.requestPermission().then(response => {
        console.log(response)
    }).catch(console.error);
```

Just need some way to remember this permission. Currently I need to grant it again every time the app is started.

**Comment by kuhnroyal** [comment]
I added a PR which allows to support this on iOS 15+.
I have not found a way to do this for older versions (13/14).
